hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/operations/SendEventOperation.java

```diff
@@ -41,7 +41,10 @@ public SendEventOperation(EventEnvelope eventEnvelope, int orderKey) {
     @Override
     public void run() throws Exception {
         EventServiceImpl eventService = (EventServiceImpl) getNodeEngine().getEventService();
-        eventService.executeEventCallback(new EventProcessor(eventService, eventEnvelope, orderKey));
+        Registration registration = eventService.getRegistration(eventEnvelope);
```

Comment:
It happens when remote event (initiated on other node) has arrived on destination node. Here it founds local EventService impl and reuse its existing (local) functionality.